OpenRA.Game/UtilityCommands/ExtractChromeStrings.cs

```diff
@@ -0,0 +1,126 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using OpenRA.FileSystem;
+
+namespace OpenRA.UtilityCommands
+{
+	sealed class ExtractChromeStringsCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name { get { return "--extract-chrome-strings"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("Extract translatable strings that are not yet localized and update chrome layout.")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			var modData = Game.ModData = utility.ModData;
+
+			var types = modData.ObjectCreator.GetTypes();
+			var translatableFields = types.SelectMany(t => t.GetFields())
+				.Where(f => f.HasAttribute<TranslationReferenceAttribute>()).Distinct();
+
+			foreach (var chromeLayout in modData.Manifest.ChromeLayout)
+			{
+				modData.ModFiles.TryGetPackageContaining(chromeLayout, out var chromePackage, out var chromeName);
+				var chromePath = Path.Combine(chromePackage.Name, chromeName);
+
+				var fluentPath = Path.Combine(Platform.EngineDir, "mods", modData.Manifest.Id, "languages/chrome/");
+				var fluentFile = Path.Combine(new Folder(fluentPath).Name, "en.ftl");
+				Directory.CreateDirectory(Path.GetDirectoryName(fluentPath));
+
+				using (var fluentWriter = new StreamWriter(fluentFile, append: true))
+				{
+					fluentWriter.WriteLine($"## {chromeLayout}");
+					var yamlBuilder = MiniYaml.FromFile(chromePath, false).Select(n => new MiniYamlNodeBuilder(n)).ToList();
+					FromChromeLayout(yamlBuilder, null, translatableFields.Select(t => t.Name).Distinct(), null, fluentWriter);
+
+					using (var chromeLayoutWriter = new StreamWriter(chromePath))
+						chromeLayoutWriter.WriteLine(yamlBuilder.WriteToString());
+				}
+			}
+		}
+
+#pragma warning disable CA1851 // Possible multiple enumerations of 'IEnumerable' collection
+		internal static void FromChromeLayout(List<MiniYamlNodeBuilder> nodes, MiniYamlNodeBuilder parent, IEnumerable<string> translatables, string container, StreamWriter fluentFile)
```

Comment:
```suggestion
		internal static void FromChromeLayout(List<MiniYamlNodeBuilder> nodes, MiniYamlNodeBuilder parent, string[] translatables, string container, StreamWriter fluentFile)
```

and then call `.ToArray()` in line 53 to silence the warning.